### PR TITLE
[Rev 1 Docs] [VnV Plan] Employee Management Tests

### LINKED ISSUE #558: [Rev 1 Docs] [VnV Plan] Functional Requirements Unit Tests - Employee Management Module
**Report:** Verification and Validation Plan
**Section Number:** 6.2

### Rubric Sections
**VnV Report Rubric**
- VnV Report Matches VnV Plan

**Final Doc Rubric**
- Update VnV Plan and Report

### PR DESCRIPTION
**Pull request by joyxiao99**
Close https://github.com/arkinmodi/project-sayyara/issues/558
Update system tests with most updated requirements
Add unit tests 

Removed system test **FRT-BE26-1** because not applicable anymore. The old **FRT-BE26-2** is now the new **FRT-BE26-1**

### CODE REVIEW

File: docs/VnVPlan/VnVPlan.tex [hunk 1]
```diff
@@ -1013,33 +1014,35 @@ \subsubsection{Employee Management}
...
 	\item \textbf{FRT-BE24-1}
 
-	      Control: Manual
+	      Type: Manual
 
-	      Initial State: Shop owner account with no employees invited to shop yet
+	      Initial State: Shop Owner has a shop profile
 
-	      Input: Shop owner enters a list of employee emails to invite
+	      Input: Employee signs up using valid information
```
Comment (LeonSo7) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      Input: Employee signs up using valid user sign up information and shop ID
```

File: docs/VnVPlan/VnVPlan.tex [hunk 2]
```diff
@@ -33,48 +33,49 @@
...
+	November 2, 2022 & Joy Xiao              & Add reflection                                             \\
+	November 2, 2022 & Leon So               & Add reflection                                             \\
+	November 2, 2022 & Timothy Choy          & Add reflection                                             \\
+	March 1, 2023    & Arkin Modi            & Add Unit Tests for User Module                             \\
+	March 1, 2023    & Arkin Modi            & Add Unit Tests for Quotes Module                           \\
+	March 1, 2023    & Arkin Modi            & Add Unit Tests for Appointments Module                     \\
+	March 1, 2023    & Arkin Modi            & Add Unit Tests for Work Orders Module                      \\
+	March 1, 2023    & Arkin Modi            & Add Unit Tests for Services Module                         \\
+	March 3, 2023    & Arkin Modi            & Add Unit Tests for Nonfunctional Requirements              \\
+	March 4, 2023    & Arkin Modi            & Update Employee Management System Tests and add Unit Tests \\
```
Comment (arkinmodi) [inline docs/VnVPlan/VnVPlan.tex]: Thanks for the credit

Comment (joyxiao99) [reply docs/VnVPlan/VnVPlan.tex]: LMAOOOO

File: docs/VnVPlan/VnVPlan.tex [hunk 3]
```diff
@@ -1013,33 +1014,35 @@ \subsubsection{Employee Management}
...
 
-	      Test Case Derivation: The system shall allow the user to enter employee email(s) to invite; the
-	      system shall send an invitation email to the invited employee(s)
+	      Test Case Derivation: The system shall allow the employee to create an account with the information
+	      provided
 
-	      How test will be performed: The tester will use a shop owner account to send employee invitation
-	      emails
+	      How test will be performed: The tester will use a shop ID displayed on the shop owner's employee
+	      profile page and go through the sign up flow as an employee. The tester will confirm that the
```
Comment (LeonSo7) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      management page and go through the sign up flow as an employee. The tester will confirm that the
```

File: docs/VnVPlan/VnVPlan.tex [hunk 4]
```diff
@@ -1013,33 +1014,35 @@ \subsubsection{Employee Management}
 
 	\item \textbf{FRT-BE24-1}
 
-	      Control: Manual
+	      Type: Manual
```
Comment (arkinmodi) [inline docs/VnVPlan/VnVPlan.tex]: For consistency we should keep this as control
```suggestion
	      Control: Manual
```

File: docs/VnVPlan/VnVPlan.tex [hunk 5]
```diff
@@ -1013,33 +1014,35 @@ \subsubsection{Employee Management}
...
-	      How test will be performed: The tester will use a shop owner account to send employee invitation
-	      emails
+	      How test will be performed: The tester will use a shop ID displayed on the shop owner's employee
+	      profile page and go through the sign up flow as an employee. The tester will confirm that the
+	      employee account has the appropriate privileges
 
 	\item \textbf{FRT-BE24-2}
 
-	      Control: Manual
+	      Type: Manual
```
Comment (arkinmodi) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      Control: Manual
```

Comment (LeonSo7) [inline docs/VnVPlan/VnVPlan.tex]: invalid user information and invalid shopID -- update to be more specific

File: docs/VnVPlan/VnVPlan.tex [hunk 6]
```diff
@@ -2500,6 +2487,99 @@ \subsubsection{Work Orders Module}
...
+
+\begin{enumerate}
+
+	\item \textbf{FRT-M7-1}
+
+	      Type: Functional, Dynamic, Unit, Automated
+
+	      Initial State: A shop and a shop owner exist
+
+	      Input: Create employee account request with valid information
```
Comment (LeonSo7) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      Input: Create employee account request with valid user information and ShopID
```

File: docs/VnVPlan/VnVPlan.tex [hunk 7]
```diff
@@ -2500,6 +2487,99 @@ \subsubsection{Work Orders Module}
...
+
+	      How test will be performed: Jest will send a create employee account request with valid information
+
+	\item \textbf{FRT-M7-2}
+
+	      Type: Functional, Dynamic, Unit, Automated
+
+	      Initial State: A shop and a shop owner exist
+
+	      Input: Create employee account request with invalid information
```
Comment (LeonSo7) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      Input: Create employee account request with invalid user information and ShopID
```

**Review by LeonSo7 — CHANGES_REQUESTED** [review]

File: docs/VnVPlan/VnVPlan.tex [hunk 8]
```diff
@@ -2500,6 +2487,99 @@ \subsubsection{Work Orders Module}
...
+
+	      Initial State: A shop account exists with employees
+
+	      Input: Get employees with an invalid shop ID
+
+	      Output: Request is rejected
+
+	      Test Case Derivation: A shop owner with employees wants to view the list of employees
+
+	      How test will be performed: Jest will send a get employees request with a invalid shop ID
```
Comment (arkinmodi) [inline docs/VnVPlan/VnVPlan.tex]:
```suggestion
	      How test will be performed: Jest will send a get employees request with an invalid shop ID
```

**Review by LeonSo7 — APPROVED** [review]

**Review by arkinmodi — APPROVED** [review]

**Pull request merged**